app/src/main/java/tech/pauly/findapet/data/AnimalRepository.kt

```diff
@@ -39,5 +36,17 @@ constructor(private val animalService: AnimalService,
                 .compose(observableHelper.applySingleSchedulers())
                 .toObservable()
     }
+
+    open fun fetchAnimal(id: Int): Observable<SingleAnimalResponse> {
```

Comment:
Actually don't use this in this PR anymore as we populate the data for favorite animals completely from cache data, but not going to delete it as we'll use it in the next pr.